services/assets/stake.go

```diff
@@ -23,40 +23,53 @@ func requestValidatorsInfo(coin coin.Coin) (AssetValidators, error) {
 }
 
 func GetValidatorsMap(api blockatlas.StakeAPI) (blockatlas.ValidatorMap, error) {
-	validators, err := GetValidators(api)
+	assets, validators, err := GetValidators(api)
 	if err != nil {
 		return nil, err
 	}
-	return validators.ToMap(), nil
+	results := normalizeValidators(assets, validators, api.Coin(), false)
+	return results.ToMap(), nil
 }
 
-func GetValidators(api blockatlas.StakeAPI) (blockatlas.StakeValidators, error) {
+func GetActiveValidators(api blockatlas.StakeAPI) (blockatlas.StakeValidators, error) {
+	assets, validators, err := GetValidators(api)
+	if err != nil {
+		return nil, err
+	}
+	results := normalizeValidators(assets, validators, api.Coin(), true)
+	return results, nil
+}
+
+func GetValidators(api blockatlas.StakeAPI) (AssetValidators, blockatlas.ValidatorPage, error) {
 	assetsValidators, err := requestValidatorsInfo(api.Coin())
 	if err != nil {
-		return nil, errors.E(err, "unable to fetch validators list from the registry").PushToSentry()
+		return nil, nil, errors.E(err, "unable to fetch validators list from the registry").PushToSentry()
 	}
 
 	validators, err := api.GetValidators()
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
-	results := normalizeValidators(validators, assetsValidators, api.Coin())
-	sort.Slice(results, func(i, j int) bool {
-		return results[i].Details.Reward.Annual > results[j].Details.Reward.Annual
-	})
-	return results, nil
+	return assetsValidators, validators, nil
 }
 
-func normalizeValidators(validators []blockatlas.Validator, assets AssetValidators, coin coin.Coin) blockatlas.StakeValidators {
+func normalizeValidators(assets AssetValidators, validators []blockatlas.Validator, coin coin.Coin, onlyActive bool) blockatlas.StakeValidators {
```

Comment:
can we remove `onlyActive` from this function? if that's possible. function itself should not control this option

for where it's needed just do filter by `onlyActive`.

Reply:
we need to create another normalize function only for active validator and other for all because  this method this is the only one knows about the asset and the parameter disabled, this parameter doesn't to the top level

Reply:
Maybe we should expose that status somehow? what other solutions other there?

Reply:
Create a new parameter inside the StakeValidator to pass this to the top lvl and filter that

Reply:
or we can remove the disabled validators before normalize

Reply:
now looks good!